### PR TITLE
API Handler for changing component state

### PR DESCRIPTION
**Pull request by hjaensch7**
Provide API for website to change component statue. Right now this means paused/active

### CODE REVIEW

File: database/database.go [hunk 1]
```diff
@@ -226,3 +226,16 @@ func (driver *Driver) FindActiveComponents() ([]worker.ComponentPath, error) {
 
 	return activeComponents, nil
 }
+
+func (driver *Driver) SetComponentStatus(compID worker.ComponentPath, status string) error {
+	updateQuery := `
+	UPDATE components SET deployment_intention = $1
```
Comment (Others) [inline database/database.go]: Start on previous line

File: handlers/pauseHandler.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+)
+
+type APIHandler struct {
+	driver        *database.Driver
+	actionManager *deployment.ActionManager
+}
+
+type APIBody struct {
+	ID            worker.ComponentPath `json:"id"`
+	UpdatedStatus string               `json:"updatedStatus"`
```
Comment (Others) [inline handlers/pauseHandler.go]: `new_deployment_intention`

File: handlers/pauseHandler.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"v9_deployment_manager/database"
+	"v9_deployment_manager/deployment"
+	"v9_deployment_manager/log"
+	"v9_deployment_manager/worker"
+)
+
+type APIHandler struct {
```
Comment (Others) [inline handlers/pauseHandler.go]: DeploymentIntentionHandler

File: handlers/pauseHandler.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+	driver        *database.Driver
+	actionManager *deployment.ActionManager
```
Comment (Others) [inline handlers/pauseHandler.go]: swapy

File: handlers/pauseHandler.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	log.Info.Println(string(body))
+	var p APIBody
+	err = json.Unmarshal(body, &p)
+	if err != nil {
+		log.Error.Println("Failed to unmarshal body", err)
+		return
+	}
+	log.Info.Println(p.ID, p.UpdatedStatus)
+	// Update Database
+	err = h.driver.SetComponentStatus(p.ID, p.UpdatedStatus)
```
Comment (Others) [inline handlers/pauseHandler.go]: `SetDeploymentIntention`

File: handlers/pauseHandler.go [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+	if err != nil {
+		log.Error.Println("Failed to update status on database", err)
+		return
+	}
+	// Notify Action Manager
+	h.actionManager.NotifyComponentStateChanged()
+	// Send Response
+	fmt.Fprintf(w, "10/4 Good Buddy")
```
Comment (Others) [inline handlers/pauseHandler.go]: Set 200?

File: main.go [hunk 7]
```diff
@@ -72,6 +72,7 @@ func main() {
 	actionManager.NotifyComponentStateChanged()
 
 	http.Handle("/payload", handlers.NewPushHandler(actionManager, driver))
+	http.Handle("/api", handlers.NewAPIHandler(actionManager, driver))
```
Comment (Others) [inline main.go]: `/api/set_deployment_intention`

File: handlers/pauseHandler.go [hunk 8]
```diff
@@ -0,0 +1,55 @@
+package handlers
```
Comment (Others) [inline handlers/pauseHandler.go]: rename file

**Review by Others — CHANGES_REQUESTED** [review]
lots of naming stuff, otherwise looks good

File: handlers/pauseHandler.go (hunk 4 above)
Comment (hjaensch7) [reply handlers/pauseHandler.go]: ?

**Review by Others — APPROVED** [review]

**Pull request merged**